### PR TITLE
markDeleted/unmarkDeleted

### PR DESCRIPTION
**Pull request by samek**
I've added markDeleted and unmarkDeleted which were missing in the java/com_spotify_voyager_jni_Index.cpp
Tests  were  also added for this addition.

### CODE REVIEW

**Review by dylanrb123 — APPROVED** [review]
looks good to me, thanks for the PR!

**Comment by samek** [comment]
1) is there something else that I need to do - I see some of the checks are not successful (I did fix lints). 
2) I could squeeze in another commit with Java_com_spotify_voyager_jni_Index_resizeIndex also implemented and tested - and we would have all three which were TODO done.

File: java/src/test/java/com/spotify/voyager/jni/IndexTest.java [hunk 1]
```diff
@@ -239,6 +239,27 @@ private void runTestWith(
...
+      } catch (Exception e) {
+        assert (true);
+      }
+      // Now let's unmark it
+      try {
+        index.unmarkDeleted(0);
+        assert (true);
+      } catch (Exception e) {
+        assert (false);
+      }
```
Comment (dylanrb123) [inline java/src/test/java/com/spotify/voyager/jni/IndexTest.java:262]: Could you update this test to not rely on exceptions as flow control and to use JUnit assertions? Would be better to mark the item as deleted, try to retrieve it, and assert that it is not returned. Similar for the `unmarkDeleted` case.

**Review by dylanrb123 — CHANGES_REQUESTED** [review]
Looking into the build failure, not 100% what's going on there. Also on second look, the tests could use some tweaks to better align with the rest of the assertions we have in the test suite.

File: java/src/test/java/com/spotify/voyager/jni/IndexTest.java (hunk 1 above)
Comment (samek) [reply java/src/test/java/com/spotify/voyager/jni/IndexTest.java:262]: Sure.

**Comment by markkohdev** [comment]
Hey @samek sorry about the radio silence, we haven't forgotten about you!  Just a lot going on as we come up to the end of the year.  We're looking into why these GPG failures are happening and will hopefully have this merged soon

**Pull request merged**